.mvn/maven.config

```diff
@@ -0,0 +1 @@
+-Pconsume-incrementals
```

Comment:
Normally this would also have `-Pmight-produce-incrementals`, but after investigating https://github.com/jenkinsci/warnings-ng-plugin/pull/58#issuecomment-489985594 I removed that line as a workaround.

The issue in that comment does appear to be caused _indirectly_ by using incrementals, though not by `incrementals-maven-plugin` itself. It looks like what is happening is that `maven-checkstyle-plugin` uses the project base directory to resolve relative paths (`etc/checkstyle-configuration.xml`). However, incrementals activates the [`might-produce-incrementals`](https://github.com/jenkinsci/plugin-pom/blob/bd80ff0f5fc89e25e265f717e17aee205415e0ee/pom.xml#L1336) Maven profile, which uses `flatten-maven-plugin` to [create an incrementals-suitable POM](https://github.com/jenkinsci/plugin-pom/blob/bd80ff0f5fc89e25e265f717e17aee205415e0ee/pom.xml#L1355-L1356) and [sets this POM as the POM for the project](https://github.com/jenkinsci/plugin-pom/blob/bd80ff0f5fc89e25e265f717e17aee205415e0ee/pom.xml#L1344), which appears to implicitly change the project's base directory to `target/` (based on discussions in flatten-maven-plugin, see https://github.com/mojohaus/flatten-maven-plugin/issues/53 and related issues), which means that `etc/checkstyle-configuration.xml` is nowhere to be found. I _think_ the root issue is [MNG-6405](https://issues.apache.org/jira/browse/MNG-6405), for which a [fix](https://github.com/apache/maven/pull/225) has been merged but not released.

I think we can remove this proprety and retain the ability to use incremental dependencies, but this plugin itself will not have incremental versions published, which seems like an ok short-term compromise until a new version of Maven is released.

Reply:
Thinking about this some more, we might need to revert the changes to pom.xml that switch the version number to `${revision}${changelist}` since we aren't intending to produce incrementals, but now that the downstream changes have been released it's probably easier to just back out the incrementalification and handle it separately so we can make sure we fully understand the problem.

Reply:
That is ok for me. 

My plan was to use (analysis-model) and produce incrementals anyway some time in the future so we can stash these changes until the maven problem has been released (and I find some spare time to enable that in the analysis-model projects as well).